### PR TITLE
Runtime interceptor & threat model loading

### PR DESCRIPTION
**Pull request by daniellovell**
Load interceptor and threat models at runtime 
Threats are now ThreatClasses
- FixedWingThreat (e.g. ucav.json)
- RotaryWingThreat (e.g. quadcopter.json 

Interceptors are now InterceptorClasses
- CarrierInterceptor (e.g. hydra70.json)
- MissileInterceptor (e.g. micromissile.json)

This update allows some interesting things for interceptors. Such as, you could launch ESSMs and Hydra 70s+micromissiles. Or something ridiculous like launch ESSMs from a Hydra 70. Can add model files for Sea Sparrows, etc.

Threats still do not utilize their model files. TODO. 

For an example of multi-threat simulation scenarios, see: `Assets/StreamingAssets/Configs/2_salvo_2_hydra_7_quad_3_ucav.json`

### CODE REVIEW

**Review by tryuan99 — APPROVED** [review]
Instead of duplicating `threadModel`/ `interceptorModel`, I wonder if it's easier to just have an `agentModel`.
Ditto with `interceptorClass`/`threatClass`

File: Assets/Scripts/Config/SimulationConfig.cs [hunk 1]
```diff
@@ -112,7 +114,7 @@ public class TargetConfig {
 [JsonConverter(typeof(StringEnumConverter))]
 public enum InterceptorType { HYDRA_70, MICROMISSILE }
 [JsonConverter(typeof(StringEnumConverter))]
-public enum ThreatType { DRONE, ANTISHIP_MISSILE }
+public enum ThreatClass { NONE, FIXEDWING, ROTARYWING, BALLISTIC}
```
Comment (tryuan99) [inline Assets/Scripts/Config/SimulationConfig.cs:117]: nit: trailing space

File: Assets/StreamingAssets/Configs/3_salvo_10_hydra_200_drones.json [hunk 2]
```diff
@@ -143,9 +140,9 @@
   "threat_swarm_configs": [
     {
       "num_agents": 200,
-      "agent_config": {
-        "interceptor_type": "HYDRA_70",
-        "threat_type": "DRONE",
+      "dynamic_agent_config": {
+        "interceptor_model": "hydra70.json",
```
[the remark this answers is not in the document]
Comment (daniellovell) [reply Assets/StreamingAssets/Configs/3_salvo_10_hydra_200_drones.json]: Good idea. Done

File: Assets/Scripts/Config/SimulationConfig.cs (hunk 1 above)
Comment (daniellovell) [reply Assets/Scripts/Config/SimulationConfig.cs]: There has been a clang-format rapture.

**Pull request merged**